### PR TITLE
sql/catalog/bootstrap,*: allow system table IDs to be dynamic

### PR DESCRIPTION
**Pull request by ajwerner**
This change is relatively small in terms of logic changes but it has large
implications for tests. The crux of the change is that we want to allow system
tables to not specify their IDs explicitly in their definition. The reason for
this is that we only reserved IDs 10-49 before handing out IDs to users. This
has the implication that new system tables in excess of 49 could overlap with
existing user descriptors. To cope with that, we'll need to use the descriptor
ID generator we normally use for user descriptors also for new system
descriptors in migrations.

Now, that could be the end of the story. The problem is that we really like
using data driven tests and we rather like printing out our keys, which
include descriptor IDs. If we were to carry on without changing the point
at which we started generating user descriptors, we'd run into real pain
every time we try to add another system table. All those datadriven tests
would have to change (and some other non-datadriven tests too).

This commit goes through that pain so hopefully for many years nobody else
has to. It does so by moving the first user descriptor we'll generate in a
newly bootstrapped cluster (or tenant) up to 100. This constant is not exported
and we can add mechanisms to override it later. This constant is chosen because
it still falls into the magical realm where the keys remain the same size.
It's also rather aesthetic. We could go bigger, but we'd pay a price for the
key for the first table you introduce. This probably doesn't matter, but I
don't want to be the one to do it.

Release note: None

### CODE REVIEW

**Comment by ajwerner** [comment]
This PR should be ready to go, though it lacks an end-to-end test. I'm inclined to add that in follow-up as keeping this thing from skewing is getting painful. I've got another branch where I'm adding the new `tenant_settings` table and will get that one passing now. cc @RaduBerinde

**Review by postamar — COMMENTED** [review]
Well, this must have been quite annoying to work through! Kudos. It's mostly good.

> Reviewed 156 of 156 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-:-Mv50Icj3AfdGTwkrszg:b-ql3jxv)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner and @RichardJCai)

___
*[pkg/keys/constants.go, line 469 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4nsPdAcbcpsaC4fyr:-Mv4nsPdAcbcpsaC4fys:b-3dehkl) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/keys/constants.go#L469)):*
> ```Go
> //
> // TODO(ajwerner): There is no reason at all for these to have their own
> // splits.
> ```

My personal impression was that we weren't being super principled in defining splits within the reserved range and it was all quite ad-hoc and spread out or duplicated in many areas of the code base. 

I'm glad this PR improves things in that regard somewhat.
___
*[pkg/migration/migrations/descriptor_utils.go, line 33 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4pwD-7z8dnZEjcAZY:-Mv4pwD-7z8dnZEjcAZZ:b7eio9y) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/migration/migrations/descriptor_utils.go#L33)):*
> ```Go
> ) error {
> 	// We install the table at the KV layer so that we can choose a known ID in
> 	// the reserved ID space. (The SQL layer doesn't allow this.)
> ```

Should this comment be updated?
___
*[pkg/migration/migrations/descriptor_utils.go, line 35 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4q7K7099Y2y2ksxhP:-Mv4q7K7099Y2y2ksxhQ:b-ufbssz) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/migration/migrations/descriptor_utils.go#L35)):*
> ```Go
> 	// the reserved ID space. (The SQL layer doesn't allow this.)
> 	err := db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
> 		tKey := catalogkeys.MakePublicObjectNameKey(codec, desc.GetParentID(), desc.GetName())
> ```

suggestion: `tKey := catalogkeys.EncodeNameKey(codec, desc)` ?
___
*[pkg/migration/migrations/descriptor_utils.go, line 54 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4qmmDEUG4SXV3TYAc:-Mv4qmmDEUG4SXV3TYAd:b-decigc) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/migration/migrations/descriptor_utils.go#L54)):*
> ```Go
> 			descProto := desc.DescriptorProto()
> 			descpb.SetID(descProto, id)
> 			desc = descbuilder.NewBuilder(descProto).BuildCreatedMutable().(catalog.TableDescriptor)
> ```

You might not need `SetID` at all here:
```
mut := tabledesc.NewBuilder(desc.TableDesc()).BuildCreatedMutable()
mut.ID = id
desc = mut
```

This comment also applies at the other `SetID` call site I saw.
___
*[pkg/sql/catalog/descs/validate.go, line 156 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4pWEe2polkLAZoOxq:-Mv4pWEe2polkLAZoOxr:b-ck00oj) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/catalog/descs/validate.go#L156)):*
> ```Go
> 	switch req.ParentID {
> 	case descpb.InvalidID:
> 		if req.ParentID == descpb.InvalidID &&
> ```

FYI, this is implied by the two lines above.
___
*[pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/full_translate, line 70 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4t6IG1Fku8rg6I_m7:-Mv4t6IG1Fku8rg6I_m8:b-4xqpwg) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/full_translate#L70)):*
> ```
> /Table/4{6-7}                              database system (host)
> /Table/4{7-8}                              database system (host)
> /Table/109{-/PrefixEnd}                    range default
> ```

Is this expected?

I noticed there were other changes in this pattern in other test cases too.
___
*[pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/misc, line 81 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4tCZpBiKU4V8OhmzO:-Mv4tCZpBiKU4V8OhmzP:b-y8jlqi) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/misc#L81)):*
> ```
> translate id=57
> ----
> /Table/5{7-8}                              range default
> ```

Shouldn't this test case and the two above have their ids bumped by 50?
___
*[pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/protectedts, line 69 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4tbJK0eE80N4rWcB4:-Mv4tbJK0eE80N4rWcB5:bbrkupa) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/protectedts#L69)):*
> ```
> 
> protect record-id=3 ts=3
> descs 56
> ```

I suspect that this test is now nonsensical because these commands (which I don't understand) refer to IDs which no longer exist.

This comment applies to other tests in this package.
___
*[pkg/sql/logictest/testdata/logic_test/crdb_internal, line 462 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4uLrv5Ui79B_lYs4E:-Mv4uLrv5Ui79B_lYs4F:bff716e) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/logictest/testdata/logic_test/crdb_internal#L462)):*
> ```
> 27  TABLE system.public.replication_stats
> 45  TABLE system.public.tenant_usage
> 54  DATABASE test
> ```

Is this expected?
___
*[pkg/sql/logictest/testdata/logic_test/create_index, line 307 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4uPsfAOdNozDzK-_Z:-Mv4uPsfAOdNozDzK-__:b-1f2van) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/logictest/testdata/logic_test/create_index#L307)):*
> ```
> SELECT status FROM [SHOW JOB WHEN COMPLETE (SELECT job_id FROM crdb_internal.jobs WHERE description LIKE 'CREATE INDEX pauseidx%')];
> ----
> failed
> ```

Fix this?
___
*[pkg/sql/opt/exec/execbuilder/testdata/lookup_join_spans, line 873 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4vBl35wfDgzUVA_Pc:-Mv4vBl35wfDgzUVA_Pd:b-t0v75r) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_spans#L873)):*
> ```
>   ol_o_id BETWEEN 20 AND 100
> ----
> Scan /Table/59/2/6463/{20-101}
> ```

Probably need to fix that WHERE clause here.
___
*[pkg/sql/opt/exec/execbuilder/testdata/schema_change_in_txn_nonmetamorphic, line 61 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4vIzH10vr26H0sGGR:-Mv4vIzH10vr26H0sGGS:b-ya3zwd) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/schema_change_in_txn_nonmetamorphic#L61)):*
> ```
> Del /Table/56/1/102/1/1
> CPut /Table/56/1/202/0 -> /TUPLE/2:2:Decimal/1000.2/1:3:Bytes/empty
> CPut /Table/56/1/202/1/1 -> /TUPLE/4:4:Decimal/0.00/2:6:Decimal/1202.2
> ```

This and the two above need fixing.
___
*[pkg/sql/opt/exec/execbuilder/testdata/secondary_index_column_families_nonmetamorphic, line 477 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4vQO_6lgePCkwkCop:-Mv4vQO_6lgePCkwkCoq:b-7a2164) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/secondary_index_column_families_nonmetamorphic#L477)):*
> ```
> UPDATE t SET z = 5 where y = 5
> ----
> CPut /Table/63/2/5/1/0 -> /BYTES/0x330a1308 (replacing [3 51 6 19 8], if exists)
> ```

These also all need fixing. Anything with `prefix=/Table/` it seems.
___
*[pkg/sql/opt/exec/execbuilder/testdata/select_index, line 1334 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4v_K18mg488BTRb7r:-Mv4v_K18mg488BTRb7s:b-1f2vb4) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/select_index#L1334)):*
> ```
> WHERE message LIKE 'Scan /Table/70/1%'
> ----
> Scan /Table/70/1/{1-2}
> ```

Fix this.
___
*[pkg/sql/testdata/telemetry/error, line 34 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4wvyd6fBq5iqmG689:-Mv4wvyd6fBq5iqmG68A:b-83sjoh) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/testdata/telemetry/error#L34)):*
> ```
>                )
> SELECT crdb_internal.unsafe_upsert_descriptor(id, crdb_internal.json_to_pb('desc', d)), d
>   FROM updated;
> ```

Nice!
___
*[pkg/ccl/backupccl/restore_old_versions_test.go, line 533 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4xTHa7Ay13xtShIFN:-Mv4xTHa7Ay13xtShIFO:b-w3rlm1) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/backupccl/restore_old_versions_test.go#L533)):*
> ```Go
> 		// These should all be updated to explicit public schemas.
> 		sqlDB.CheckQueryResults(t, "SELECT * FROM system.namespace WHERE name='public' AND id = 29", [][]string{
> 			{"1", "0", "public", "29"},
> ```

This is no longer checking that there exist schemas named "public" which have id != 29 iff parentID != 1. Can I suggest you replace this query with
```
SELECT
	(id = 29) AS is_system_schema, count(*)
FROM
	system.namespace
WHERE
	"parentSchemaID" = 0 AND name = 'public'
GROUP BY
	is_system_schema
ORDER BY
	is_system_schema;
```
or something like this which I feel is more in line with the spirit of the original without relying on non-reserved ID values.
___
*[pkg/ccl/cliccl/debug_backup_test.go, line 67 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4z7W0C50rxD5PrLZP:-Mv4z7W17iEKgYRTTHWG:bix1iyr) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/cliccl/debug_backup_test.go#L67)):*
> ```Go
> 	sqlDB.Exec(t, `BACKUP DATABASE testDB TO $1 AS OF SYSTEM TIME `+ts2.AsOfSystemTime(), backupPath)
> 
> 	makeExpander := func(t *testing.T, extra [][2]string) func(string) string {
> ```

This might be worth making a proper test util for the benefit of other packages as well, perhaps? It's neat.
___
*[pkg/sql/tests/repair_test.go, line 330 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv5-WDK1UpgMaViaipg:-Mv5-WDK1UpgMaViaiph:b-9an0ww) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/tests/repair_test.go#L330)):*
> ```Go
> 		}
> 	}
> 	for caseIdx, tc := range []testCase{
> ```

I appreciate what you did in this test file but this just reminds me that we should rethink these repair tests entirely, they're so brittle. Super useful, but brittle.

**Review by ajwerner — COMMENTED** [review]
One of these modestly hurt, and the translator things I just didn't feel up to making that agnostic right now. I did fix it though. 

I think this should be about ready to go. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-:-Mv5_iCO2J2foUfEaj8o:b-kgtlvo)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @postamar, and @RichardJCai)

___
*[pkg/migration/migrations/descriptor_utils.go, line 33 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4pwD-7z8dnZEjcAZY:-Mv59Iiy5nVe3JrQFCCC:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/migration/migrations/descriptor_utils.go#L33)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Should this comment be updated?
</blockquote></details>

Done.
___
*[pkg/migration/migrations/descriptor_utils.go, line 35 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4q7K7099Y2y2ksxhP:-Mv59INHAC3DV6cPWsie:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/migration/migrations/descriptor_utils.go#L35)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

suggestion: `tKey := catalogkeys.EncodeNameKey(codec, desc)` ?
</blockquote></details>

Done.
___
*[pkg/migration/migrations/descriptor_utils.go, line 54 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4qmmDEUG4SXV3TYAc:-Mv59ZV86o6lpqK9gh40:bs7wumx) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/migration/migrations/descriptor_utils.go#L54)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

You might not need `SetID` at all here:
```
mut := tabledesc.NewBuilder(desc.TableDesc()).BuildCreatedMutable()
mut.ID = id
desc = mut
```

This comment also applies at the other `SetID` call site I saw.
</blockquote></details>

Done. 
___
*[pkg/sql/catalog/descs/validate.go, line 156 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4pWEe2polkLAZoOxq:-Mv59s0cC-0ivoGpitKJ:b-gtgjzu) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/catalog/descs/validate.go#L156)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

FYI, this is implied by the two lines above.
</blockquote></details>

ah yeah, whoops, reverted
___
*[pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/full_translate, line 70 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4t6IG1Fku8rg6I_m7:-Mv5HEZe1Gz5PeMe0s0H:bebuxq2) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/full_translate#L70)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Is this expected?

I noticed there were other changes in this pattern in other test cases too.
</blockquote></details>

It's because prefix-end for `/Table/109` takes an extra byte of `0xFF` instead of just being `/Table/110` because this is the discontinuity in encoding size for the varint encoder: https://github.com/cockroachdb/cockroach/blob/c38023e5350b191fae4ea7f36d561d3b309212d2/pkg/util/encoding/encoding.go#L121
___
*[pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/misc, line 81 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4tCZpBiKU4V8OhmzO:-Mv5KfWyB781wCc9a7LM:b-yl8pgk) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/misc#L81)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Shouldn't this test case and the two above have their ids bumped by 50?
</blockquote></details>

I started to try to add some let functionality here but it was too painful and I just changed it to the right values.
___
*[pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/protectedts, line 69 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4tbJK0eE80N4rWcB4:-Mv5L5IwEM7syiJrKgBE:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/protectedts#L69)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

I suspect that this test is now nonsensical because these commands (which I don't understand) refer to IDs which no longer exist.

This comment applies to other tests in this package.
</blockquote></details>

Done.
___
*[pkg/sql/logictest/testdata/logic_test/crdb_internal, line 462 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4uLrv5Ui79B_lYs4E:-Mv5MvFq1FqOiCalNwAK:b-wyj6sw) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/logictest/testdata/logic_test/crdb_internal#L462)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Is this expected?
</blockquote></details>

Added some let var, it's fixed now
___
*[pkg/sql/logictest/testdata/logic_test/create_index, line 307 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4uPsfAOdNozDzK-_Z:-Mv5Wu8tCTaoHiZz6XHW:b-c3iom6) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/logictest/testdata/logic_test/create_index#L307)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Fix this?
</blockquote></details>

That was painful. Done. 
___
*[pkg/sql/opt/exec/execbuilder/testdata/lookup_join_spans, line 873 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4vBl35wfDgzUVA_Pc:-Mv5YXUd5rnR5I4iEjNX:b6p4ns6) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_spans#L873)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Probably need to fix that WHERE clause here.
</blockquote></details>

Right idea. Done. 
___
*[pkg/sql/opt/exec/execbuilder/testdata/schema_change_in_txn_nonmetamorphic, line 61 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4vIzH10vr26H0sGGR:-Mv5YuYc-yolMZidAVdO:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/schema_change_in_txn_nonmetamorphic#L61)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

This and the two above need fixing.
</blockquote></details>

Done.
___
*[pkg/sql/opt/exec/execbuilder/testdata/secondary_index_column_families_nonmetamorphic, line 477 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4vQO_6lgePCkwkCop:-Mv5_dQn5R6-J4TWTUjR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/secondary_index_column_families_nonmetamorphic#L477)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

These also all need fixing. Anything with `prefix=/Table/` it seems.
</blockquote></details>

Done.
___
*[pkg/sql/opt/exec/execbuilder/testdata/select_index, line 1334 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4v_K18mg488BTRb7r:-Mv5ZEtx8uK4phhDsqFb:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d325d75b51b531bec606482a90a46e41802ac17c/pkg/sql/opt/exec/execbuilder/testdata/select_index#L1334)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Fix this.
</blockquote></details>

Done.
___
*[pkg/sql/testdata/telemetry/error, line 34 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4wvyd6fBq5iqmG689:-Mv5_g301A3nH3wmhYLF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/testdata/telemetry/error#L34)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

Nice!
</blockquote></details>

Done.
___
*[pkg/ccl/backupccl/restore_old_versions_test.go, line 533 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4xTHa7Ay13xtShIFN:-Mv5_9zyCKrl_BmNLeEV:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/backupccl/restore_old_versions_test.go#L533)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

This is no longer checking that there exist schemas named "public" which have id != 29 iff parentID != 1. Can I suggest you replace this query with
```
SELECT
	(id = 29) AS is_system_schema, count(*)
FROM
	system.namespace
WHERE
	"parentSchemaID" = 0 AND name = 'public'
GROUP BY
	is_system_schema
ORDER BY
	is_system_schema;
```
or something like this which I feel is more in line with the spirit of the original without relying on non-reserved ID values.
</blockquote></details>

Done.
___
*[pkg/ccl/cliccl/debug_backup_test.go, line 67 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4z7W0C50rxD5PrLZP:-Mv5_FZT5Tdz8MpjnGAP:bw0tj2d) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/cliccl/debug_backup_test.go#L67)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

This might be worth making a proper test util for the benefit of other packages as well, perhaps? It's neat.
</blockquote></details>

I'm going to hold off on that. It's a bit ad-hoc, but it is a useful tool. 
___
*[pkg/sql/tests/repair_test.go, line 330 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv5-WDK1UpgMaViaipg:-Mv5_S0T7S9jGa07NfC7:b-8fx2u7) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/sql/tests/repair_test.go#L330)):*
<details><summary><i>Previously, postamar (Marius Posta) wrote…</i></summary><blockquote>

I appreciate what you did in this test file but this just reminds me that we should rethink these repair tests entirely, they're so brittle. Super useful, but brittle.
</blockquote></details>

:+1:

**Review by ajwerner — COMMENTED** [review]
Err forgot to get rid of SetID, but that's gone now too 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-:-Mv5byDd-Wdlv8BAyKGs:bw8f4za)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @postamar, and @RichardJCai)

**Review by postamar — APPROVED** [review]
:lgtm_strong: 

:ship: :it: 

> Reviewed 11 of 12 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-:-Mv5e6Bn3_dFdA5oIMZM:b-fz38bi)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajwerner and @RichardJCai)

___
*[pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/full_translate, line 70 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76003#-Mv4t6IG1Fku8rg6I_m7:-Mv5eNoO-4h3AatHjcDr:bb741sf) ([raw file](https://github.com/cockroachdb/cockroach/blob/c814a3e10dbeeae91387b71b9e419c899310852e/pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/full_translate#L70)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

It's because prefix-end for `/Table/109` takes an extra byte of `0xFF` instead of just being `/Table/110` because this is the discontinuity in encoding size for the varint encoder: https://github.com/cockroachdb/cockroach/blob/c38023e5350b191fae4ea7f36d561d3b309212d2/pkg/util/encoding/encoding.go#L121
</blockquote></details>

TIL!

**Comment by RaduBerinde** [comment]
Thanks for working on this! :tada:

File: pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/tenant/protectedts [hunk 1]
```diff
@@ -36,26 +36,26 @@ descs 54
 
 translate database=db
 ----
-/Tenant/10/Table/5{6-7}                    num_replicas=7 num_voters=5 pts=[1 2]
```
Comment (adityamaru) [inline pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/tenant/protectedts:39]: hmm just noticed that this PR seems to have removed the pts fields that we expect to be populated post translation. Will investigate on monday.

Comment (adityamaru) [reply pkg/ccl/spanconfigccl/spanconfigsqltranslatorccl/testdata/tenant/protectedts:39]: The desc IDs needed changing, picked it up in - https://github.com/cockroachdb/cockroach/pull/76606

**Pull request merged**